### PR TITLE
 Visitor can see different products on a restaurants menu 

### PR DESCRIPTION
**Pull request by mantanquzzmo**
[ PT ](https://www.pivotaltracker.com/story/show/170152669)

- adds menu with products
-passes both features and unit tests

Skärmavbild 2019-12-10 kl  15 50 34

### CODE REVIEW

File: app/views/products/index.html.erb [hunk 1]
```diff
@@ -1,3 +1,8 @@
-<h1>Products#index</h1>
-<p>Find me in app/views/products/index.html.erb</p>
+<h1>Greta's Top Picks</h1>
+<% @products.each do |product| %>
+  <%= product.name %>
+  <%= product.desc %>
+  <%= product.price %>
+  <% end %>
```
Comment (tochman) [inline app/views/products/index.html.erb:10]: Please be mindful of code formatting

File: db/migrate/20191210122304_create_products.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateProducts < ActiveRecord::Migration[6.0]
+  def change
+    create_table :products do |t|
+      t.string :name
+      t.string :desc
```
Comment (tochman) [inline db/migrate/20191210122304_create_products.rb]: you should go with `description` Clarity is good.

File: features/step_definitions/background_step.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Given("the following products exist:") do |table|
+  table.hashes.each do |attrs|
```
Comment (tochman) [inline features/step_definitions/background_step.rb]: `attrs` is okay. `product_attributes` is better. Clarity is good

File: app/views/products/index.html.erb (hunk 1 above)
Comment (tochman) [reply app/views/products/index.html.erb:10]: I would use `<ul>` and `<li>` in this case to create a better structure. You can always remove it later when you add styling.

**Review by tochman — CHANGES_REQUESTED** [review]
Good job. Small comments

File: app/views/products/index.html.erb (hunk 1 above)
Comment (mantanquzzmo) [reply app/views/products/index.html.erb:10]: Agree

File: db/migrate/20191210122304_create_products.rb (hunk 2 above)
Comment (mantanquzzmo) [reply db/migrate/20191210122304_create_products.rb]: Ok

File: features/step_definitions/background_step.rb (hunk 3 above)
Comment (mantanquzzmo) [reply features/step_definitions/background_step.rb]: forgot about this one. 1 sec.

File: app/views/products/index.html.erb (hunk 1 above)
Comment (tochman) [reply app/views/products/index.html.erb:10]: ...and format code. Again

File: spec/models/product_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_presence_of :price } 
+  end
+
+  describe 'Factory' do
+    it 'should have valid Factory' do
+      expect(FactoryBot.create(:product)).to be_valid
```
Comment (tochman) [inline spec/models/product_spec.rb]: Remove `FactoryBot` No need for that since you include the matchers in `rails_helper`

**Review by oliverochman — APPROVED** [review]

**Pull request merged**